lib/fastlane/actions/s3.rb

```diff
@@ -254,6 +262,39 @@ def self.upload_plist_and_html_to_s3(s3_access_key, s3_secret_access_key, s3_buc
         Helper.log.info "Successfully uploaded ipa file to '#{Actions.lane_context[SharedValues::S3_IPA_OUTPUT_PATH]}'".green
       end
 
+      def self.upload_dsym(s3_access_key, s3_secret_access_key, s3_bucket, s3_region, dsym_file_name, dsym_file_data)
+        Actions.verify_gem!('aws-sdk')
+        require 'aws-sdk'
+        if s3_region
+          s3_client = AWS::S3.new(
+            access_key_id: s3_access_key,
+            secret_access_key: s3_secret_access_key,
+            region: s3_region
+          )
+        else
+          s3_client = AWS::S3.new(
+            access_key_id: s3_access_key,
+            secret_access_key: s3_secret_access_key
+          )
+        end
+
+        bucket = s3_client.buckets[s3_bucket]
+
+        dsym_obj = bucket.objects.create(dsym_file_name, dsym_file_data, acl: :public_read)
+
+        # When you enable versioning on a S3 bucket,
+        # writing to an object will create an object version
+        # instead of replacing the existing object.
+        # http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/S3/ObjectVersion.html
+        if dsym_obj.kind_of? AWS::S3::ObjectVersion
+          dsym_obj = dsym_obj.object
+        end
+
+        # Setting actionand environment variables
```

Comment:
This comment was already there i just copied it to another method (:
Anyway, incoming update